### PR TITLE
closes #2523 add more examples for derive_var_joined_exist_flag

### LINKED ISSUE #2523: Documentation: Enhance documentation for `derive_var_joined_exist_flag()`
## Background

derive_var_joined_exist_flag() can be a challenging function to use - more simple examples will help show its power and ease of use.  The example below is a nice showcase.  The SAS code can be included as that is our user base (for now!) to bring in more understanding.

```R
library(tibble)

ex <- tribble(
  ~STUDYID, ~USUBJID, ~EXDOSE, ~EXSTDTM,
  
  "STUDY001", "SUBJ001", 2, "2024-01-01T08:00",
  "STUDY001", "SUBJ001", 4, "2024-01-02T08:00",
  
  "STUDY001", "SUBJ002", 2, "2024-01-01T08:30",
  "STUDY001", "SUBJ002", 4, "2024-01-02T08:30",
  "STUDY001", "SUBJ002", 2, "2024-01-03T08:30",   # Down-titration
  "STUDY001", "SUBJ002", 2, "2024-01-04T08:30",
  
  "STUDY001", "SUBJ003", 2, "2024-01-01T09:00",
  "STUDY001", "SUBJ003", 4, "2024-01-03T09:00",
  "STUDY001", "SUBJ003", 4, "2024-01-04T09:00",
  "STUDY001", "SUBJ003", 8, "2024-01-05T09:00",
  "STUDY001", "SUBJ003", 4, "2024-01-06T09:00",   # Down-titration
  
  "STUDY001", "SUBJ004", 2, "2024-01-01T10:00",
  "STUDY001", "SUBJ004", 4, "2024-01-02T10:00",
  "STUDY001", "SUBJ004", 4, "2024-01-03T10:00"
)

ex_flag <- ex %>%
  # Flag each EXDOSE which is lower than previous per patient
  derive_var_joined_exist_flag(
    dataset_add = ex,
    by_vars = exprs(STUDYID, USUBJID),
    order = exprs(EXSTDTM),
    new_var = DOSE_REDUCED,
    tmp_obs_nr_var = tmp_dose_nr,
    join_vars = exprs(EXDOSE),
    join_type = "before",
    first_cond_lower = NULL,
    first_cond_upper = NULL,
    filter_add = NULL,
    filter_join = (
      tmp_dose_nr == tmp_dose_nr.join + 1   # Look only at adjacent doses (i.e. next to each other)
      & EXDOSE > 0 & EXDOSE.join > 0        # Both doses are valid
      & EXDOSE < EXDOSE.join                # Dose is lower than previous
    )
  )
```

```SAS
data ex_flag (drop = prev_dose);
    set ex;
    by USUBJID EXSTDTM;
    
    if first.USUBJID then call missing(prev_dose);
    else prev_dose = lag(EXDOSE);
    
    if 0 < EXDOSE < prev_dose < 0 then DOSE_REDUCED = "Y";
run;
```

_Originally posted by @yurovska in https://github.com/pharmaverse/admiralmetabolic/issues/31#issuecomment-2394528279_

## Definition of Done

* Documentation for this function is updated with this example.  
~~* SAS pseudo Code is also provided in example~~
* The arguments with NULL can be removed to increase brevity.

### PR DESCRIPTION
**Pull request by rossfarrugia**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Review the [Cheat Sheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf). Make any required updates to it by editing the file `inst/cheatsheet/admiral_cheatsheet.pptx` and re-upload a PDF and a PNG version of it to the same folder. (The PNG version can be created by taking a screenshot of the PDF version.)
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers). A Developer Notes section is available in `NEWS.md` for tracking developer-facing issues.  
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by rossfarrugia** [comment]
This one doesn't warrant a NEWS entry in my mind.

**Comment by bms63** [comment]
I think it is nice to mention in the news!!  It is also reminds us to write it up in the blog

File: R/derive_var_joined_exist_flag.R [hunk 1]
```diff
@@ -406,6 +407,58 @@
...
+#'
+#' # flag each dose which is lower than the previous dose per subject
+#' ex <- tribble(
+#'   ~USUBJID, ~EXSTDTM,          ~EXDOSE,
+#'   "1",      "2024-01-01T08:00",      2,
+#'   "1",      "2024-01-02T08:00",      4,
+#'   "2",      "2024-01-01T08:30",      1,
+#'   "2",      "2024-01-02T08:30",      4,
+#'   "2",      "2024-01-03T08:30",      3,
+#'   "2",      "2024-01-04T08:30",      2
```
Comment (bundfussr) [inline R/derive_var_joined_exist_flag.R:419]: I would add a record to show that only the adjacent dose is considered.
```suggestion
#'   "2",      "2024-01-04T08:30",      2,
#'   "2",      "2024-01-05T08:30",      2,
```

File: R/derive_var_joined_exist_flag.R [hunk 2]
```diff
@@ -406,6 +407,58 @@
...
+#'   adqs,
+#'   dataset_add = adqs,
+#'   new_var = NODDFL,
+#'   by_vars = exprs(USUBJID, PARAMCD),
+#'   join_vars = exprs(CHGCAT1),
+#'   join_type = "after",
+#'   order = exprs(ADY),
+#'   filter_join = CHGCAT1.join != "Worsened" # flags if followed by a non-deterioration
+#' ) %>%
+#'   mutate(DDETERFL = if_else(CHGCAT1 == "Worsened" & is.na(NODDFL), "Y", NA_character_))
```
Comment (bundfussr) [inline R/derive_var_joined_exist_flag.R:461]: This example could be simplified:
```suggestion
#' derive_var_joined_exist_flag(
#'   adqs,
#'   dataset_add = adqs,
#'   new_var = DDETERFL,
#'   by_vars = exprs(USUBJID, PARAMCD),
#'   join_vars = exprs(CHGCAT1),
#'   join_type = "all",
#'   order = exprs(ADY),
#'   filter_join = all(CHGCAT1.join == "Worsened" | ADY > ADY.join)
#' )
```

Comment (rossfarrugia) [reply R/derive_var_joined_exist_flag.R:461]: Took me a minute to get my head around, but yeh it works and less lines and saves calling dplyr

**Review by bundfussr — APPROVED** [review]

**Comment by yurovska** [comment]
> @yurovska just checking if these new examples from @rossfarrugia (TYTY) would of assuaged some of your frustrations with admiral!?

I wouldn’t say I was ever frustrated. {admiral} is awesome! It was more about me facing some challenges due to my lack of experience. These examples definitely help, though - thanks!

**Review by yurovska — APPROVED** [review]

**Comment by bms63** [comment]
The prodigal son returns 

image

**Pull request merged**